src/field_info.rs

```diff
@@ -297,3 +303,41 @@ impl SetterSettings {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+pub struct Transform {
+    pub params: Vec<(syn::Pat, syn::Type)>,
+    pub body: syn::Expr,
+}
+
+fn parse_transform_closure(expr: &syn::Expr) -> Result<Transform, Error> {
+    let closure = match expr {
+        syn::Expr::Closure(closure) => closure,
+        _ => return Err(Error::new_spanned(expr, "Expected closure")),
+    };
+    if let Some(kw) = &closure.asyncness {
+        return Err(Error::new(kw.span, "Transform closure cannot be async"));
+    }
+    if let Some(kw) = &closure.capture {
+        return Err(Error::new(kw.span, "Transform closure cannot capture variables"));
+    }
+
+    let inputs = &closure.inputs;
+    let mut params = Vec::new();
+    for input in inputs {
```

Comment:
This can be done with a `collect`.

Reply:
I used a loop because I didn't want to deal with `collect::<Result>()`. Could change it if it's a problem for you.